### PR TITLE
Release 2024-04-29

### PR DESCRIPTION
**Pull request by aurangzaib048**

### CODE REVIEW

**Comment by bcaller** [comment]
@SocketSecurity ignore pypi/alembic

(not sure if that works)

The SQLi issues above can be ignored as they are within migrations, but I don't have permissions to click resolve.

**Comment by diracdeltas** [comment]
@SocketSecurity ignore pypi/alembic@1.13.1

File: db/alembic/versions/2024_03_05_1334-ec0e245c0a9e_initialize_schema_and_pk_generation.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                    result := result | (app_db_id << 14);
+
+                    -- use the remaining 14 bits to store an identifier
+                    -- that's unique to this millisecond. That's where the
+                    -- 16384 comes from (2**14) for calculating seq_id.
+                    result := result | (seq_id);
+                end;
+                $$ language plpgsql;
+            """  # nosec
+        )
```
[the remark this answers is not in the document]
Comment (thypon) [reply db/alembic/versions/2024_03_05_1334-ec0e245c0a9e_initialize_schema_and_pk_generation.py]: fine

File: db/alembic/versions/2024_03_05_1334-ec0e245c0a9e_initialize_schema_and_pk_generation.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    )
+
+
+def downgrade() -> None:
+    conn = op.get_bind()
+    conn.execute(text(f"DROP SEQUENCE {main_config.schema_name}.id_seq;"))
```
[the remark this answers is not in the document]
Comment (thypon) [reply db/alembic/versions/2024_03_05_1334-ec0e245c0a9e_initialize_schema_and_pk_generation.py]: fine

File: db/alembic/env.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        exist already.
+        :return: None
+        """
+        connection.execute(
+            text(
+                f"""
+                CREATE SCHEMA IF NOT EXISTS {main_config.schema_name};
+                SET search_path TO {main_config.schema_name};
+                """
+            )
```
[the remark this answers is not in the document]
Comment (thypon) [reply db/alembic/env.py:62]: fine

File: db/alembic/versions/2024_03_05_1334-ec0e245c0a9e_initialize_schema_and_pk_generation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (thypon) [reply db/alembic/versions/2024_03_05_1334-ec0e245c0a9e_initialize_schema_and_pk_generation.py:65]: fine

**Review by porteron — APPROVED** [review]

**Pull request merged**